Ginger/GingerCore/Actions/ActLaunchJavaWSApplication.cs

```diff
@@ -550,70 +553,71 @@ private bool ValidateArguments()
                 {
-                    Error = "The Java path folder '" + mJavaWSEXEPath_Calc + "' is not valid, please select the Java Bin folder.";
+                    Error = $"The Java path folder '{mJavaWSEXEPath_Calc}' is not valid, please select the Java Bin folder.";
                     return false;
                 }
 
-                if (mLaunchJavaApplication == true)
+                if (mLaunchJavaApplication)
                 {
-                    bool isValidURL = SetURLExtensionType(mURL_Calc);
-                    if (!isValidURL)
+                    if (!SetURLExtensionType(mURL_Calc))
                     {
-                        Error = "The provided Java application path '" + mURL_Calc + "' is not valid, please select .Jar or .Jnlp file or .Exe file";
+                        Error = $"The provided Java application path '{mURL_Calc}' is not valid, please select .Jar or .Jnlp file or .Exe file";
                         return false;
                     }
                 }
 
-                if ((mLaunchJavaApplication == true && WaitForWindowWhenDoingLaunch == true) || mLaunchWithAgent == true)
+                if ((mLaunchJavaApplication && WaitForWindowWhenDoingLaunch) || mLaunchWithAgent)
                 {
                     if (string.IsNullOrEmpty(mWaitForWindowTitle_Calc))
                     {
                         Error = "Missing valid Java application Window title to search for.";
                         return false;
                     }
 
-                    if (string.IsNullOrEmpty(mWaitForWindowTitleMaxTime_Calc) || int.TryParse(mWaitForWindowTitleMaxTime_Calc, out mWaitForWindowTitleMaxTime_Calc_int) == false)
+                    if (string.IsNullOrEmpty(mWaitForWindowTitleMaxTime_Calc) || !int.TryParse(mWaitForWindowTitleMaxTime_Calc, out mWaitForWindowTitleMaxTime_Calc_int))
                     {
                         Error = "Max waiting time for java application window is not valid.";
                         return false;
                     }
                 }
 
-                if (mLaunchWithAgent == true)
+                if (mLaunchWithAgent)
                 {
-                    if (string.IsNullOrEmpty(mAttachAgentProcessSyncTime_Calc) || int.TryParse(mAttachAgentProcessSyncTime_Calc, out mAttachAgentProcessSyncTime_Calc_int) == false)
+                    if (string.IsNullOrEmpty(mAttachAgentProcessSyncTime_Calc) || !int.TryParse(mAttachAgentProcessSyncTime_Calc, out mAttachAgentProcessSyncTime_Calc_int))
                     {
                         Error = "Process sync time for attach agent is not valid.";
                         return false;
                     }
 
-                    if (Directory.Exists(mJavaAgentPath_Calc) == false || File.Exists(Path.Combine(mJavaAgentPath_Calc, "GingerAgent.jar")) == false)
+                    bool agentFilesExist = Directory.Exists(mJavaAgentPath_Calc) &&
```

Comment:
Please check this agent file exist on the basis of version that will be selected.